garage/np/algos/batch_polopt.py

```diff
@@ -1,135 +1,84 @@
-from dowel import logger, tabular
+from collections import deque
 
-from garage.experiment import snapshotter
+from dowel import tabular
+import numpy as np
+
+from garage.misc import special
 from garage.np.algos.base import RLAlgorithm
-from garage.plotter import Plotter
-from garage.sampler import BatchSampler
 
 
 class BatchPolopt(RLAlgorithm):
     """
-    Base class for batch sampling-based policy optimization methods.
-    This includes various policy gradient methods like vpg, npg, ppo, trpo,
-    etc.
+    Base class for cem and cam_es.
+
+    Args:
+        env_spec (garage.envs.EnvSpec): Environment specification.
+        policy (garage.tf.policies.base.Policy): Policy.
+        baseline (garage.tf.baselines.Baseline): The baseline.
+        n_samples(int): Number of policies sampled in one epoch.
+        gae_lambda(float): Lambda used for generalized advantage estimation.
+        discount (float): Discount.
+        max_path_length (int): Maximum length of a single rollout.
     """
 
     def __init__(self,
-                 env,
+                 env_spec,
                  policy,
                  baseline,
-                 scope=None,
-                 n_itr=500,
-                 start_itr=0,
-                 batch_size=5000,
-                 max_path_length=500,
-                 discount=0.99,
+                 n_samples,
```

Comment:
were all of these constructor arguments unused?

Reply:
Seems so. The current implementation of cem and cam_es on master has all these arguments though.

Reply:
if an argument is not used in a class, it should not be in the constructor. if instead, those arguments are used by child classes, the arguments should appear in the child constructor, not the parent constructor.

can you remove the arguments not used by `BatchPolopt`, and add them to CEM and CMAES as needed?